include/wlr/types/wlr_xdg_shell_v6.h

```diff
@@ -32,6 +32,77 @@ struct wlr_xdg_client_v6 {
 	struct wl_event_source *ping_timer;
 };
 
+enum wlr_positioner_v6_anchor {
+	/**
+	 * the center of the anchor rectangle
+	 */
+	WLR_POSITIONER_V6_ANCHOR_NONE = 0,
+	/**
+	 * the top edge of the anchor rectangle
+	 */
+	WLR_POSITIONER_V6_ANCHOR_TOP = 1,
+	/**
+	 * the bottom edge of the anchor rectangle
+	 */
+	WLR_POSITIONER_V6_ANCHOR_BOTTOM = 2,
+	/**
+	 * the left edge of the anchor rectangle
+	 */
+	WLR_POSITIONER_V6_ANCHOR_LEFT = 4,
+	/**
+	 * the right edge of the anchor rectangle
+	 */
+	WLR_POSITIONER_V6_ANCHOR_RIGHT = 8,
+};
```

Comment:
-1, can't we just use the protocol enums?

Reply:
bump

Reply:
We should be consistent. We already redefine enums for other protocols.

Reply:
I'll give an opinion, but these fields are never required to be read by rootston so the stakes aren't very high here.

This serves several functions. We probably won't include the generated header with the library, so the wrapper type ensures we have a type. Types are better than just using uint32_t because they serve as documentation without people needing to look up what the values mean in the protocol, or be required to use wayland-scanner themselves to generate the header. This makes the library more accessible and user friendly. It could end up saving our users hours of time.

Defining the enums adds a trivial amount of complexity to the library. There are no moving parts, and these values should never change (they will bump the version if they make breaking changes). So if you do the cost/benefit analysis, it seems to me that defining the enums is a win by adding value to the user.

This can be seen as the responsibility of the library to do because really at the end of the day, isn't everything we have done here just a collection of wrapper types just like this?

Reply:
I agree, but let's not merge more the wrong way

Reply:
@acrisci 

I don't want to duplicate code. There is a single source of truth for these values, and it is the protocol extensions. Getting wayland-scanner working is the *least* of your worries when you're making a Wayland compositor based on wlroots.

![untitled](https://user-images.githubusercontent.com/1310872/38061168-429e28f0-32bc-11e8-8e7e-79a0d65b1825.png)

Reply:
![sava](https://user-images.githubusercontent.com/506932/38061306-c555af2a-32bc-11e8-8a5f-a326f2595c29.png)